### DISCUSSION TITLE
Loading GFS data - cfgrib engine?

### DISCUSSION THREAD

**Discussion by GunnarQuante** [contrailcirrus/pycontrails#88] [Q&A] [answered]
When running the example for downloading GFS data "Run Cocip with GFS data" ([https://py.contrails.org/examples/GFS.html]), I receive an error message (s. below) I have pycontrails[gfs] installed and already tried pip install eccodes, pip install cfgrib and to import each of them. 

Am I missing some other packages/dependencies here or is there something missing in the gfs_met.open_metdataset() file? 
```

  Cell In[3], line 24
    met = gfs_met.open_metdataset()

  File ~\.conda\envs\traffic\lib\site-packages\pycontrails\datalib\gfs\gfs.py:420 in open_metdataset
    self.download(**xr_kwargs)

  File ~\.conda\envs\traffic\lib\site-packages\pycontrails\core\datalib.py:531 in download
    self.download_dataset(times_to_download)

  File ~\.conda\envs\traffic\lib\site-packages\pycontrails\datalib\gfs\gfs.py:381 in download_dataset
    self._download_file(t)

  File ~\.conda\envs\traffic\lib\site-packages\pycontrails\datalib\gfs\gfs.py:498 in _download_file
    ds = self._open_gfs_dataset(temp_grib_filename, t)

  File ~\.conda\envs\traffic\lib\site-packages\pycontrails\datalib\gfs\gfs.py:544 in _open_gfs_dataset
    tmpds = xr.open_dataset(

  File ~\.conda\envs\traffic\lib\site-packages\xarray\backends\api.py:556 in open_dataset
    backend = plugins.get_backend(engine)

  File ~\.conda\envs\traffic\lib\site-packages\xarray\backends\plugins.py:205 in get_backend
    raise ValueError(

ValueError: unrecognized engine cfgrib must be one of: ['netcdf4', 'h5netcdf', 'scipy', 'store', 'zarr']
```

**Comment by GunnarQuante** [comment]
Seems like I found it! Apparently, the ECMWFLibs package ([https://pypi.org/project/ecmwflibs/]) is also required

**Reply by zebengberg** [reply, marked as answer]
Thanks for figuring this out @GunnarQuante !

Getting these ECMWF libraries installed is not easy. I do have `cfgrib` installed without `ecmwflibs`, but it required that I first installed the binary `eccodes` library.

Note that [ECMWF](https://github.com/ecmwf/cfgrib#installation) suggests installing with conda to avoid building the binaries: `conda install -c conda-forge cfgrib`.

The quick `python -m cfgrib selfcheck` is a nice way to check if your environment is configured correctly.

We can publish a Dockerfile or a Colab notebook showing best practice for getting these system dependencies installed for `cfgrib` or [`metview`](https://metview.readthedocs.io/en/latest/).
